session/session.go

```diff
@@ -2877,6 +2877,21 @@ func (s *session) GetTxnWriteThroughputSLI() *sli.TxnWriteThroughputSLI {
 	return &s.txn.writeSLI
 }
 
+var _ telemetry.TemporaryTableFeatureChecker = &session{}
+
+// TemporaryTableExists is used by the telemetry package to avoid circle dependency.
+func (s *session) TemporaryTableExists() bool {
```

Comment:
How often is it called? It seems time-consuming if it's called frequently.

Reply:
Every 6 hours...

Reply:
https://github.com/pingcap/tidb/blob/6e9d2750ae2042fb01acabb44de627ba92d7a1f4/domain/domain.go#L1054
and https://github.com/pingcap/tidb/blob/6e9d2750ae2042fb01acabb44de627ba92d7a1f4/telemetry/telemetry.go#L39